### PR TITLE
Implement tagging for the Lambda function

### PR DESCRIPTION
**Pull request by cristim**
# Issue Type

- Feature Pull Request

## Summary
This makes the CloudFormation stack more human-friendly by converting it to YAML, and implements tagging for the Lambda function in both CloudFormation and Terraform.

You can see #247 for more context.

## Code contribution checklist

1. [x] The contribution fixes a single existing github issue, and it is linked
   to it.
1. [x] The code is as simple as possible, readable and follows the idiomatic Go
  [guidelines](https://golang.org/doc/effective_go.html).
1. [x] All new functionality is covered by automated test cases so the overall
  test coverage doesn't decrease.
1. [x] No issues are reported when running `make full-test`.
1. [x] Functionality not applicable to all users should be configurable.
1. [x] Configurations should be exposed through Lambda function environment
   variables which are also passed as parameters to the
   [CloudFormation](https://github.com/cristim/autospotting/blob/master/cloudformation/stacks/AutoSpotting/template.json)
   and
   [Terraform](https://github.com/cristim/autospotting/blob/master/terraform/autospotting.tf)
   stacks defined as infrastructure code.
1. [x] Global configurations set from the infrastructure stack level should also
   support per-group overrides using tags.
1. [x] Tags names and expected values should be similar to the other existing
   configurations.
1. [x] Both global and tag-based configuration mechanisms should be tested and
  proven to work using log output from various test runs.
1. [x] The logs should be kept as clean as possible (use log levels as
  appropriate) and formatted consistently to the existing log output.
1. [x] The documentation is updated to cover the new behavior, as well as the
   new configuration options for both stack parameters and tag overrides.
1. [ ] A code reviewer reproduced the problem and can confirm the code
  contribution actually resolves it.

### CODE REVIEW

**Comment by cristim** [comment]
Still todo: make sure the docs are pointing to the YAML stack code.

File: cloudformation/stacks/AutoSpotting/parameters.yaml [hunk 1]
```diff
@@ -1,17 +1,9 @@
...
+LambdaZipPath: nightly/lambda.zip
 MinOnDemandNumber: '1'
-
-# Valid choices:
-# - for EC2 Classic environments: Linux/UNIX | SUSE Linux | Windows
-# - for VPC: Linux/UNIX (Amazon VPC) | SUSE Linux (Amazon VPC) | Windows (Amazon VPC)
-SpotProductDescription: "Linux/UNIX (Amazon VPC)"
\ No newline at end of file
+SpotProductDescription: Linux/UNIX (Amazon VPC)
+TagFilteringMode: opt-out
```
Comment (xlr-8) [inline cloudformation/stacks/AutoSpotting/parameters.yaml]: Why removing the comments?

Comment (xlr-8) [inline cloudformation/stacks/AutoSpotting/parameters.yaml]: Why removing the comments though?

File: cloudformation/stacks/AutoSpotting/template.yaml [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        State: "ENABLED"
+        Targets:
+          -
+            Arn:
+              Fn::GetAtt:
+                - "LambdaFunction"
+                - "Arn"
+            Id: "AutoSpottingEventGenerator"
+      Type: "AWS::Events::Rule"
+
```
Comment (xlr-8) [inline cloudformation/stacks/AutoSpotting/template.yaml]: As JSON is a subset of YAML, I assume you made an automatic conversion?

File: cloudformation/stacks/AutoSpotting/parameters.yaml (hunk 1 above)
Comment (cristim) [reply cloudformation/stacks/AutoSpotting/parameters.yaml]: It was done automatically by the tool that I used to convert the stack code.

I will bring them back.

File: cloudformation/stacks/AutoSpotting/template.yaml (hunk 2 above)
Comment (cristim) [reply cloudformation/stacks/AutoSpotting/template.yaml]: Yes, I used `json2yaml template.json > template.yaml`

**Pull request merged**